virttest/utils_env.py

```diff
@@ -314,69 +234,44 @@ def get_lvmdev(self, name):
-                         (port, username, address))
-
-            self._tcpdump = aexpect.ShellSession(
-                login_cmd,
-                output_func=_update_address_cache,
-                output_params=(self,))
-
-            remote.handle_prompts(self._tcpdump, username, password, prompt)
-            self._tcpdump.sendline(cmd)
-
-        else:
-            cmd = cmd_template % utils_path.find_command("tcpdump")
-            self._tcpdump = aexpect.Tail(command=cmd,
-                                         output_func=_tcpdump_handler,
-                                         output_params=(self, "tcpdump.log"))
-
-        if not self._tcpdump.is_alive():
-            logging.warn("Could not start tcpdump")
-            logging.warn("Status: %s", self._tcpdump.get_status())
-            msg = utils_misc.format_str_for_message(self._tcpdump.get_output())
-            logging.warn("Output: %s", msg)
-
-    def start_tcpdump(self, params):
-        self._params = params
-
-        if "address_cache" not in self.data:
-            self.data["address_cache"] = {}
+    def start_ip_sniffing(self, params):
+        """
+        Start ip sniffing.
 
-        if self._tcpdump is None:
-            self._start_tcpdump()
-        else:
-            if not self._tcpdump.is_alive():
-                del self._tcpdump
-                self._start_tcpdump()
-
-    def stop_tcpdump(self):
-        if self._tcpdump is not None:
-            self._tcpdump.close()
-            del self._tcpdump
-            self._tcpdump = None
+        :param params: Params object.
+        """
+        self.data.setdefault("address_cache", ip_sniffing.AddrCache())
+        sniffers = ip_sniffing.Sniffers
+
+        if not self._sniffer:
+            remote_pp = params.get("remote_preprocess") == "yes"
+            remote_opts = None
+            session = None
+            if remote_pp:
+                client = params.get('remote_shell_client', 'ssh')
+                remote_opts = (params.get['remote_node_address'],
+                               params.get('remote_shell_port', '22'),
+                               params['remote_node_user'],
+                               params['remote_node_password'],
+                               params.get('remote_shell_prompt', '#'))
+                session = remote.remote_login(client, *remote_opts)
+            for s_cls in sniffers:
+                if s_cls.is_supported(session):
+                    self._sniffer = s_cls(self.data["address_cache"],
+                                          "ip-sniffer.log",
+                                          remote_opts)
+                    break
+            if session:
+                session.close()
+
+        if not self._sniffer:
+            raise exceptions.TestError("Can't find any supported ip sniffer! "
+                                       "%s" % [s.command for s in sniffers])
+
+        if not self._sniffer.is_alive():
```

Comment:
this is not really necessary as `_sniffer.start` checks whether it's already alive... anyway no big deal...

Reply:
Sure, will modify this.